dsl/pace/dsl/stencil_config.py

```diff
@@ -63,10 +64,24 @@ def __init__(
         if communicator:
             set_distributed_caches(self)
 
+    def check_communicator(self, communicator: CubedSphereCommunicator):
+        if communicator.partitioner.layout[0] != communicator.partitioner.layout[1]:
+            raise RuntimeError(
+                "Trying to run with a non-square layout is not supported"
+            )
+        if (
+            communicator.partitioner.layout[0] > 3
+            or communicator.partitioner.layout[1] > 3
+        ) and (self.run_mode != RunMode.Run and self.use_minimal_caching):
+            raise RuntimeError(
+                "Can't compile with a layout larger than 3x3 with minimal caching on"
```

Comment:
Should this error message also cover when you didn't specify run mode as run and minimal caching?

Reply:
this is covered in https://github.com/ai2cm/pace/blob/bb5249e8814f2349203a5bf2524ccfe5c9ac8a4f/dsl/pace/dsl/stencil_config.py#L44